src/interfaces/chain.h

```diff
@@ -260,7 +260,7 @@ class Chain
     virtual void getPackageLimits(unsigned int& limit_ancestor_count, unsigned int& limit_descendant_count) = 0;
 
     //! Check if transaction will pass the mempool's chain limits.
-    virtual bool checkChainLimits(const CTransactionRef& tx) = 0;
+    virtual bool checkChainLimits(const CTransactionRef& tx, std::string& err_string) = 0;
```

Comment:
Not blocking.
Could return `util::Result<void>` and inline all the `err_string` return messages within the `CheckPackageLimits` function.

Reply:
Thanks  @furszy updated to your suggestion, its more elegant, and no for out parameter `err_string`